examples/nlp/gpt/train_gpt_dpo.py

```diff
@@ -147,6 +136,14 @@ def main(cfg) -> None:
         train_dataloader=train_dataloader,
         val_dataloader=val_dataloader,
         test_dataloader=None,
+        collate_fn=partial(
```

Comment:
Why was this necessary? What was wrong with doing the partial wrap in train_gpt_dpo rather than passing it to the algo class?

Reply:
Linking this thread since it's the same convo: https://github.com/NVIDIA/NeMo-Aligner/pull/386/files#r1838581055

I needed to make this change b/c I want to do a collective to get the max seq len (for padding) globally, but I couldn't do it in the dataloader b/c of multiprocessing. So now the padding happens outside of the dataloader